oe_theme.theme

```diff
@@ -210,153 +210,300 @@ function oe_theme_preprocess_paragraph__oe_links_block(array &$variables): void
       ],
-    ];
-  }
+      [
+        'section_id' => 2,
+        'list_class_name' => 'ecl-footer-standardised__list--inline',
+        'title_class_name' => 'ecl-footer-standardised__title--separator',
+      ],
+    ],
+    [
+      [
+        'section_id' => 3,
+        'title_class_name' => 'ecl-footer-standardised__title--separator',
+      ],
+      [
+        'section_id' => 3,
+        'title_class_name' => 'ecl-footer-standardised__title--separator',
+      ],
+    ],
+    [
+      'section_id' => 6,
+      'list_class_name' => 'ecl-footer-standardised__list--condensed',
+    ],
+    [
+      'title' => $variables['processed']['ec_site_link'],
+      'section_id' => 7,
+    ],
+    [
+      'links' => $variables['processed']['service_navigation'],
+      'section_id' => 8,
+    ],
+    [
+      'links' => $variables['processed']['legal_navigation'],
+      'section_id' => 9,
+    ],
+  ];
+}
 
-  $variables['corporate_footer']['sections'] = [
-    $sections_about_ec,
-    $sections_social_media,
-    $sections_about_eu,
+/**
+ * Implements hook_preprocess_oe_corporate_blocks_eu_footer().
+ */
+function oe_theme_preprocess_oe_corporate_blocks_eu_footer(array &$variables): void {
+  $variables['processed'] = [];
+  $corporate_footer = $variables['corporate_footer'];
+  $variables['processed']['site_name'] = \Drupal::config('system.site')->get('name');
+  $variables['processed']['site_path'] = Url::fromRoute('<front>', [], ['absolute' => TRUE])->toString();
+  $variables['processed']['site_owner'] = t('This site is managed by the @name', ['@name' => $variables['site_owner']]);
+
+  $path = Url::fromRoute('<front>', [], ['absolute' => TRUE])->toString();
+  $langcode = \Drupal::languageManager()->getCurrentLanguage()->getId();
+  $logo = $variables['ecl_logo_path'] . '/logo--' . EuropeanUnionLanguages::getInternalLanguageCode($langcode) . '.svg';
+  $logo = [
+    'path' => $path,
+    'language' => $langcode,
+    'src_mobile' => $logo,
+    'src_desktop' => $logo,
   ];
+  $variables['processed']['logo'] = $logo;
 
-  // Create the list of bottom link.
-  foreach ($corporate_footer['bottom_links'] as $item) {
-    $variables['corporate_footer']['common'][] = [
-      'link' => [
-        'label' => $item['label'],
-        'path' => $item['href'],
-      ],
-    ];
-  }
+  $variables['processed']['contacts'] = _oe_theme_preprocess_footer_links($corporate_footer['contact']);
+  $variables['processed']['social_media'] = _oe_theme_preprocess_footer_links($corporate_footer['social_media']);
+  $variables['processed']['institution_links'] = _oe_theme_preprocess_footer_links($corporate_footer['institution_links']);
+  $variables['processed']['legal_links'] = _oe_theme_preprocess_footer_links($corporate_footer['legal_links']);
 
-  $variables['custom_footer'] = _oe_theme_preprocess_oe_corporate_blocks_site_specific_footer($variables);
+  $ecl_branding = theme_get_setting('branding') ?? 'core';
```

Comment:
I think we can use $variables['ecl_branding'] . Value should be available.

Reply:
done but see above